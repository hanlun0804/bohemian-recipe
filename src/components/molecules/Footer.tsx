import React from 'react'

const Footer: React.FC = () => {
    return (
        <footer className="bottom-0 left-0 w-full bg-black h-10 flex items-center justify-center">
            <p className="text-orange-300 text-xs">
                &copy; {new Date().getFullYear()} Bohemian Recipe
            </p>
        </footer>
    )
}

export default Footer
