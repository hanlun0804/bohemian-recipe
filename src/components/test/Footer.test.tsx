import { describe, expect, it } from 'vitest'
import { render, screen } from '@testing-library/react'
import '@testing-library/jest-dom'
import Footer from '../molecules/Footer'

describe('Footer', () => {
    it('Renders Bohemian Recipe', () => {
        render(<Footer />)
        const currentYear = new Date().getFullYear()
        const expectedText = `${currentYear} Bohemian Recipe`

        expect(screen.getByText(expectedText))
        expect(document.body).toMatchSnapshot()
    })
})
