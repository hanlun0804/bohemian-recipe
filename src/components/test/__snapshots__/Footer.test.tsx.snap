// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Footer > Renders Bohemian Recipe 1`] = `
<body>
  <div>
    <footer
      class="bottom-0 left-0 w-full bg-black h-10 flex items-center justify-center"
      data-testid="cypress-footer"
    >
      <p
        class="text-orange-300 text-xs"
      >
        2023
         Bohemian Recipe
      </p>
    </footer>
  </div>
</body>
`;
